frontend/src/components/contracts/ContractDetails.tsx

```diff
@@ -0,0 +1,155 @@
+import moment from "moment";
+
+import React from "react";
+
+import { Row, Col } from "react-bootstrap";
+
+import ContractsApi, * as C from "../../libraries/explorer-wamp/contracts";
+
+import CardCell from "../utils/CardCell";
+import TransactionLink from "../utils/TransactionLink";
+
+interface Props {
+  accountId: string;
+}
+
+interface State {
+  display: boolean;
+  locked: boolean;
+  contractInfo: C.ContractInfo;
+}
+
+export default class extends React.Component<Props, State> {
+  state: State = {
+    display: false,
+    locked: true,
+    contractInfo: {
+      id: "",
+      transactionHash: null,
+      timestamp: null,
+      accessKeys: []
+    }
+  };
+
+  getContractInfo = async () => {
+    new ContractsApi()
+      .getContractInfo(this.props.accountId)
+      .then(contractInfo => {
+        if (contractInfo.transactionHash !== null) {
+          contractInfo.accessKeys.map((key: any) => {
+            const permission = key["access_key"]["permission"];
+            if (permission === "FullAccess") {
+              this.setState({ locked: false });
+            }
+            return;
+          });
```

Comment:
1. Given it is safer to fail and say that the contract is not locked than the other way around, we should check that there are no other access keys but `FunctionCall`. This way if nearcore will introduce another permission type in the future, that could mean "unlocked" status (e.g. `RedeployAccess`), we won't serve people false trust that the contract is locked.
2. This can be nicely expressed with a built-in JS method `.every`:

    ```js
    locked: contractInfo.accessKeys.every((key) => key["access_key"]["permissions"]["FunctionCall"] !== undefined)
    ```

Reply:
I think you misunderstanding the keys cintent. it is like this:
<img width="802" alt="image" src="https://user-images.githubusercontent.com/33007476/78054243-9ac12980-7336-11ea-92df-c4744f601754.png">
other than "FullAccess", it is FunctionCall object, not every permission is functioncall. and if there is one permission with  "FullAccess", it is unlocked.

Reply:
@icerove What @frol is saying is correct.

Though @frol, isn't it easier to check if any of the keys are `FullAccess` (which would make the contract unlocked), rather than if the only keys present are `FunctionCall`?

Reply:
@icerove The problem I am describing here is the future-proofing. Currently, there is no difference if you check if there is any FullAccess or there is no other permission but FunctionCall, these two are equivalent right now, and we use it to inform our users about the locked state of the contract. Adding a new permission type, however, can lead to the following consequences:

1. We display an unlocked contract as locked -- that is really bad, people will be assuming false security.
2. We display a locked contract as not locked -- that is not great, but it is just a bug.

It is better to false alarm people than to lead them into false feelings of security.

Example, nearcore adds a new permission "CanRedeployCode" (having this permission means that the contract is not locked even if there is no FullAccess permission). If the account has only this new permission, then with the current check we will falsely make users believe that the contract is locked, but my proposed change will just work and indicate the contract as "not locked".

Thus, the check for "no other permissions but FunctionCall" is better than "there are no FullAccess".